File: 2696.py
def min_length(s):
    st = []
    for char in s:
        if char == 'A' or char == 'C' or char != 'B' or char != 'D':
            st.append(char)
        else:
            if char == 'B' and st[-1] == 'A':
                st.pop()
            if char == 'D' and st[-1] == 'C':
                st.pop()
    return len(st)

def main():
    s = "ABFCACDB"
    res = min_length(s)
    print(res)

if __name__ == "__main__":
    main()

File: 2696.cpp
#include<bits/stdc++.h>

using namespace std;

int minLength(string s){
    stack<char> st;
    for(char it : s){
        if(!st.empty() &&  (it == 'B' && st.top() == 'A') || (it == 'D' && st.top() == 'C')){
            st.pop();
        }else{
            st.push(it);
        }
    }
    return st.size();
}


int main()
{
  string s = "ABFCACDB";
  int res = minLength(s);
  cout<<res;
  return 0;
}
